webapp/src/main/java/org/apache/atlas/web/security/AtlasAuthenticationProvider.java

```diff
@@ -130,6 +131,12 @@ public Authentication authenticate(Authentication authentication)
             } else if (keycloakAuthenticationEnabled) {
                 try {
                     authentication = atlasKeycloakAuthenticationProvider.authenticate(authentication);
+                } catch (KeycloakAuthenticationException ex) {
+                    if(authentication instanceof KeycloakAuthenticationToken) {
```

Comment:
1. lets check the exception instance here, not the authentication object. 
2. Also you might need to throw AtlasBaseException and throwable inside that exception would be `KeycloakAuthenticationException`, so that statement 1 holds good